client/trino-client/src/test/java/io/trino/client/auth/external/MockRedirectHandler.java

```diff
@@ -1,21 +1,42 @@
 package io.trino.client.auth.external;
 
 import java.net.URI;
+import java.time.Duration;
 
 public class MockRedirectHandler
         implements RedirectHandler
 {
     private URI redirectedTo;
+    private int redirectionCount;
+    private Duration redirectTime;
 
     @Override
     public void redirectTo(URI uri)
             throws RedirectException
     {
         redirectedTo = uri;
+        redirectionCount++;
+        try {
+            Thread.sleep(redirectTime.toMillis());
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
no throw RTE?

Reply:
I think that since this is for tests, we just want to propagate the interrupt signal to the framework calling our test and finish the tests asap. No need to throw some more exceptions I think.

Reply:
I think we should. Thread is interrupted and so it should not go back to regular processing. I guess that nobody interrupts this sleep now so it works any way.

BTW. Can you please add a tests where you interrupt getting the token to see how other threads will behave?

Reply:
The idea was good, but I can't make them not flaky. The issue is that once in a while, when a thread is interrupted, some other threads are paused to the point where the whole execution is no longer parallel, hence triggering resolving of the token twice. Due to that I think that I should skip that test for now, as I don't want to add test that is potentially flaky to the suite.